internal/config/config.go

```diff
@@ -100,7 +101,7 @@ func LoadConfig() (Config, error) {
 		return config, err
 	}
 
-	homeDir, err := homedir.Dir()
+	homeDir, err := os.UserHomeDir()
```

Comment:
Since you are replacing the library - it is being used here as well 
https://github.com/asdf-vm/asdf/pull/1970/files#diff-54c7c1af5fa8d5db4dc49f0e8e80e93ba2b1183ba4d5c9e2e5729e6deae6a3cdL207

There is a related issue 
https://github.com/asdf-vm/asdf/issues/1923